### DISCUSSION TITLE
Sorting & Repeater

### DISCUSSION THREAD

**Discussion by cupcakearmy** [directus/directus#2710] [Archive]
Hi everyone,

I have a question on how to model certain situations in Directus.
Basically I have 2 scenarios which I don't know how to handle:

1. Sort multiple images.
Imagine a collection `products` that has a field `photos`. There is the `files` field type, but one can only select multiple photos, but not decide on the order of them.

2. Two Collections. One for recipes and one for ingredients.
The problem: Assign multiple ingredients to one recipe, with the corresponding amount. 
e.g.
Recipe:
Title: Pizza
Ingredients: [ {someIngredientId, 200g}, {someOtherIgredientId, 5g}, ... ]

One way is probably to create a third collection with simply one ingredient and a text field and assign multiple of that to the recipe collection. But this seems a bit... wrong.
I know there is the [relational repeater](https://github.com/directus/app/issues/1996) coming in v9 but i don't know what the timeline on that is :)

Any suggestions/best pratices?

Thanks!!!

**Comment by cupcakearmy** [comment]
Btw, since the repeater also supports sorting, the relational repeater would solve basically both problems

**Reply by benhaynes** [reply]
No definite update on the relational repeater, but it _should_ be released in v9 (coming soon) which can be tracked on the `app-next` and `api-next` repos. I agree that this is the _right_ way to solve this... though you could create a custom interface in the meantime if needed.

**Reply by cupcakearmy** [reply]
Thanks for the quick reply! What about the sortable images? is there a way to do that without the relational repeater?

**Reply by benhaynes** [reply]
I believe you can use the `files` interface and make sure that the junction table has a `sort` field. That should work.

**Reply by cupcakearmy** [reply]
That works yes, thanks :)
